python/pyspark/sql/dataframe.py

```diff
@@ -451,10 +451,10 @@ def repartition(self, numPartitions, *cols):
         +---+-----+
         |age| name|
         +---+-----+
-        |  5|  Bob|
-        |  5|  Bob|
         |  2|Alice|
+        |  5|  Bob|
         |  2|Alice|
+        |  5|  Bob|
         +---+-----+
```

Comment:
It's not a problem. This testcase is not robust since it uses the result of hash-partitioning.
```
>>> data.explain()
16/06/18 21:25:42 DEBUG SparkOptimizer: 
=== Result of Batch Operator Optimizations ===
 RepartitionByExpression [age#0], 7    RepartitionByExpression [age#0], 7
!+- RepartitionByExpression [age#0]    +- Union
!   +- Union                              :- LogicalRDD [age#0, name#1]
!      :- LogicalRDD [age#0, name#1]      +- LogicalRDD [age#0, name#1]
!      +- LogicalRDD [age#0, name#1]   
```